odoo/addons/base/models/ir_default.py

```diff
@@ -126,16 +126,18 @@ def get_model_defaults(self, model_name, condition=False):
             current user), as a dict mapping field names to values.
         """
         cr = self.env.cr
-        query = """ SELECT f.name, d.json_value FROM ir_default d
+        query = """ SELECT f.name, d.json_value
+                    FROM ir_default d
                     JOIN ir_model_fields f ON d.field_id=f.id
-                    JOIN res_users u ON u.id=%s
                     WHERE f.model=%s
-                        AND (d.user_id IS NULL OR d.user_id=u.id)
-                        AND (d.company_id IS NULL OR d.company_id=u.company_id)
+                        AND (d.user_id IS NULL OR d.user_id=%s)
+                        AND (d.company_id IS NULL OR d.company_id=%s)
                         AND {}
                     ORDER BY d.user_id, d.company_id, d.id
                 """
-        params = [self.env.uid, model_name]
+        # prefetch_fields is necessary during module installation
+        # because the res_users table may not have all prescribed columns.
+        params = [model_name, self.env.uid, self.with_context(prefetch_fields=False).env.company.id]
```

Comment:
Is this one still necessary?  The evaluation of the `ormcache` key above should have fetched the field in cache, shouldn't it?

Reply:
Indeed